---
title: "ANGSD Project"
author: "Yuqi Cheng"
date: "2021/4/17"
output: 
  html_document: 
    toc: yes
    toc_float: yes
    toc_depth: 2
    theme: spacelab
    number_sections: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

<span style="font-size:17px;">
Androgen receptor (AR) is a survival factor for prostate cancer cells that plays an essential role in cancer progression. However, after a period of androgen deprivation treatment (ADT), cancer cells will develop resistance to castration<sup>[1]</sup>. To overcome the castration resistance, deeper insights into the reactivating factor are needed.</span>

<span style="font-size:17px;">
Efforts to understand resistance to AR-targeted therapy have focused mainly on cell-intrinsic mechanisms, especially AR signal self-reactivation, in the past few years<sup>[2,3]</sup>. But there is increasing evidence indicating that the microenvironment is a driver of drug resistance in various cancers, including prostate cancer<sup>[4,5]</sup>. In recent research, a factor called NRG1 was found to play an important role in the resistance gaining process. To learn more about NRG1, it is essential to understand how does the NRG1 reactive the cancer cells and gain an insight into the different downstream reactions between the NRG1-induced CRPCs and normal prostate cancer<sup>[6]</sup>.</span>

<span style="font-size:17px;">
In this study, we hypothesize that castration-resistant prostate cancer cells (CRPCs) induced by NRG1 have different gene expression profiles from normal prostate cancer cells (AR-regulated prostate cancer). The drug resistance is not entirely caused by the reactivation of AR-related pathways. There might be other pathways that allow cancer cells to survive, and CRPCs (induced by NRG1) have different activated functional clusters from normal prostate cancer cells that have not developed drug resistance.
</span>

# Result

## NRG1-induced CRPCs shows a big expression profile difference compared with normal prostate cancer

<span style="font-size:17px;">
Using the featureCounts result, we first examined the distance between NRG1-induced CRPCs and AR-regulated prostate cancer. We used the prostate cancer cells treated with Enzalutamide (an androgen receptor (AR) target inhibitor, Enz) to show the AR-regulated gene, the cells treated with NRG1 to show the NRG1-regulated genes and the cancer cells treated with both Enz and NRG1 to show the resistance to Enz (NRG1-induced CRPC). Samples treated with DMSO (vehicle) as control. </span>

<span style="font-size:17px;">Consistent with expectations, the NRG1 reactivated CRPCs samples (Enz+NRG1 treated) show a far distance with AR-regulated prostate cancer samples (vehicle_treated) (Figure 1A and 1B). The result indicates that there is a difference between the expression profile of NRG1-induced CRPCs and normal prostate cancer.</span>

<span style="font-size:17px;">The volcano plot show the differential expressed genes compared with normal prostate cancer. We can find that the expression levels of 11023 genes have been changed (Figure 1C), and the differential expression profile is shown in the heatmap (Figure 1D). NRG1-induced CRPCs develop a different expression profile when they gain the resistance, and obtaining such resistance is not entirely caused by the reactivation of AR-related pathways.</span>

<span style="font-size:17px;">Meanwhile, we also generated GO enrichment plot, please check supplementary [Figure S1](https://github.com/yuqcheng/ANGSD_project/tree/main/figure). The GO analysis show that the differential expressed genes mainly engage in the DNA replication process and ribosome biogenesis process. This may correlate with the malignant features of the CRPCs. </span>

```{r result1, echo=FALSE, out.width = '100%'}
knitr::include_graphics("fig1.png")
```

## NRG1 activates a subset of AR target genes

<span style="font-size:17px;">In the figure 1D, interestingly, we can also find that NRG1-induced CRPCs share a small part of the same profile with normal prostate cancer. This same part of profile may indicate that NRG1 also activates a subset of AR target genes. The differential expression analysis of the AR-regulated and NRG1-regulated genes show a co-regulation gene cluster (Figure 2). </span>

```{r result2, echo=FALSE, out.width = '100%'}
knitr::include_graphics("fig2.png")
```

<span style="font-size:17px;">From figure c, we can see that target genes of NRG1 and AR have a overlap. Then we use these 5713 overlap target gene to do clustering and get 4 clusters. In the cluster 4, we can clearly find that NRG1 activates a subset of AR target genes. </span>

## Conclusion

<span style="font-size:17px;">To sum up, the castration-resistant prostate cancer cells (CRPCs) induced by NRG1 do have different gene expression profiles from normal prostate cancer cells (AR-regulated prostate cancer), our hypothesis is correct. Meanwhile, we find that although the expression profile is different, there is a small subset of AR target genes that can be reactivated by NRG1. The drug resistance is not entirely caused by the reactivation of AR-related pathways, but there are some parts of the pathway be the same.</span>

# Method {.tabset .tabset-fade .tabset-pills}

<span style="font-size:17px;">
Brief introduction of our analysis pipeline is shown as follows,

```{r pressure, echo=FALSE, out.width = '100%'}
knitr::include_graphics("pipeline.png")
```

<span style="font-size:17px;">
Please check the following tabs for more details.</span>

## Library Preparation for RNA-seq

<span style="font-size:17px;">
1x106 22Pc-EP cells was plated in 6-well plate, growing in regular RPMI-1640 containing 10% FBS. Enzalutamide (500nM) or recombinant NRG1 peptide (10 ng/mL) was added the next day. After 48 hours, cells were trypsinized and collected by spinning at 500 g for 1.5 min at 4<sup>。</sup>C. Cells were then washed once with cold 1X PBS and spinned down at 500 g for 1.5 min at 4<sup>。</sup>C. After discarding supernatant, cells were lysed using 50 mL cold lysis buffer (10 mM Tris-HCl pH 7.4, 10 mM NaCl, 3 mM MgCl2, 0.1% IGEPAL CA-360) and spinned down immediately at 500 g for 10 min, 4<sup>。</sup>C. RNA was extracted using PureLink RNA Mini Kit (Thermo Fisher Scientific #12183025) following manufacturer’s instructions. RNA was diluted into 200 ng/mL with DEPC-treated water (Thermo Fisher Scientific #AM9916).</span>

<span style="font-size:17px;">
RNA-Seq libraries were prepared using the Illumina TruSeq stranded mRNA kit, with 10 cycles of PCR amplification, starting from 500 ng of total RNA, at the Genome Technology Center (GTC) at New York University Langone Medical Center. Barcoded RNA-Seq were run as paired-end read 50 nucleotides in length on the Illumina NovaSeq 6000 and Poly-A selection was performed. </span>

## Data Downloading

<span style="font-size:17px;">
We downloaded all data (42 pair-ended fastq.gz files) from ENA. Study Accession: **PRJNA622705**. The reason why we downloaded data from ENA rather than SRA is because the files in ENA had been spilted into pair-ended format.</span>

<span style="font-size:17px;">
Link: https://www.ebi.ac.uk/ena/browser/view/PRJNA622705
</span>

<span style="font-size:17px;">
Here, we have 4 samples (Cancer cells with different treatment), each sample has 3 replicates. 


<span style="font-size:17px;">
- **Sample 1: DMSO (treated with DMSO (vehicle))**

<span style="font-size:17px;">
- **Sample 2: Enz (treated with Enzalutamide, a androgen receptor (AR) target inhibitor)**

<span style="font-size:17px;">
- **Sample 3: NRG1 (treated with recombinant NRG1 peptide, a factor that induce CRPCs gain resistance to Enz)**

<span style="font-size:17px;">
- **Sample 4: Enz+NRG1 (treated with both Enz and NRG1 to show the resistance to Enz)**
</span>

<span style="font-size:17px;">
You can find all the fastq files in my scratch folder. Here we show the first 5 files' downloading ftp.
</span>

```{r download}
library(knitr)
data <- read.delim('filereport_read_run_PRJNA622705_tsv.txt', sep = "\t")
showdata <- data.frame(Run_Accession = data$run_accession, Fastq_ftp = data$fastq_ftp)
kable(showdata[1:5,])
```

## FastQC result

### Run fastqc

<span style="font-size:17px;">
It's very important to make sure we have fastq files of good quality, so what we need to do first is quality control before doing an alignment. Here, we used `fastqc` to do that. Because the fastq files are pair-ended, we process 2 fastq files (_1 and _2) together. Then we used `multiqc` to synthesis all the results. The bash code is shown as follows,
</span>

```{bash eval=FALSE}
#! /bin/bash

#SBATCH --partition=panda
#SBATCH --nodes=1
#SBATCH --mem=16G
#SBATCH --job-name=fastqc

source ~/.bash_profile^C
spack load fastqc

for i in {57..77}; do
fastqc -o /athena/angsd/scratch/yuc4009/fastqc -t 6 /athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_1.fastq.gz /athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_2.fastq.gz;
done;

spack load -r py-multiqc@1.7
multiqc .
```

<span style="font-size:17px;">
Finally, we downloaded the multiqc-report from the cluster via `WinSCP`. To interpret these results, we conduct some analysis in the following subsections.
</span>

### Sequence Quality

```{r sq, echo=FALSE, out.width = '100%'}
knitr::include_graphics("seq_quality.png")
knitr::include_graphics('per_seq_quality.png')
```

<span style="font-size:17px;">
From `Sequence Quality Histograms` and `Per Sequence Quality Scores`, we can find all the 42 fastq files are in good sequencing quality (both in base level or read level).</span>

### Per Sequence GC Content

<span style="font-size:17px;">
To check if there were any contaminations (e.g. other species' DNA contamination and/or adapter contamination) in the library, we need to look `Per Sequence GC content`.</span>

```{r gc, echo=FALSE, out.width = '100%'}
knitr::include_graphics("per_seq_gc_content.png")
```

<span style="font-size:17px;">
The GC content plot shows a normal distribution without an extra peak in all fastq files. We can say that our library is not contaminated and can be used for further analysis.</span>

### Sequence Duplication Levels

<span style="font-size:17px;">
The `Sequence Duplication levels` plot shows a fail results, which means there is a relatively high duplication level in our sequencing result.</span>

```{r dl, echo=FALSE, out.width = '100%'}
knitr::include_graphics("seq_dup_level.png")
```

<span style="font-size:17px;">
From the figure above, we can find that all samples are near the same duplication level. However, the fail result here is expected because we usually over amplify samples in order to detect low expression genes in transcriptome sequencing. In other words, the high duplication level is produced by the high sequencing depth. Thus, in this case, we do not need to worry about our data at hand.</span>

### Per Base N Content

<span style="font-size:17px;">
Meanwhile, we get a good `Per Base N Content` result in our fastqc report.</span>

```{r n, echo=FALSE, out.width = '100%'}
knitr::include_graphics("n_content.png")
```

### Conlusion For FastQC

<span style="font-size:17px;">
**To sum up, the fastqc report (multiqc report) shows that all fastq files we have are of good quality and can be used for alignment directly. The fail result happened in the duplication level is expected, we do not worry about our data. Moreover, the `Per sequence quality scores` and `Per Sequence GC Content` show a good quality so that it is not necessary to use `TrimGalore` to perform a trimming step.**</span>

## STAR Alignment

### STAR indexing

<span style="font-size:17px;">
After making sure the quality of our data is good enough to conduct further analysis, we used `STAR` software to do alignment. We first downloaded the `fa file` and `gtf file` to generate index of STAR. We chose `hg38.fa` and `hg38.ncbiRefSeq.gtf` as sources.</span>

<span style="font-size:17px;">
gtf file link: https://hgdownload.cse.ucsc.edu/goldenpath/hg38/bigZips/genes/hg38.ncbiRefSeq.gtf </span>

<span style="font-size:17px;">
fa file link: https://hgdownload.cse.ucsc.edu/goldenpath/hg38/bigZips/hg38.fa </span>

```{bash eval=F}
##Generating STAR index

#! /bin/bash -l

#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=star_index
#SBATCH --mem=50G

STAR --runMode genomeGenerate --runThreadN 8 --genomeDir star_index --genomeFastaFiles /athena/angsd/scratch/yuc4009/refdata/hg38.fa --sjdbGTFfile /athena/angsd/scratch/yuc4009/refdata/hg38.ncbiRefSeq.gtf --sjdbOverhang 99
```

<span style="font-size:17px;">
Then we used STAR to do the alignment. Because we have 4 samples, each sample has 6 pair-ended fastq files. To make the file storage structure clearer, we repeat the alignment work four times for 4 samples.</span>

### Run STAR (bash code) {.tabset .tabset-pills}

<span style="font-size:17px;">
All these code can be found in my [github](https://github.com/yuqcheng/ANGSD_project/tree/main/bash%20files).</span>

#### star_nrg1.bash

```{bash eval=F}
#! /bin/bash -l

#SBATCH --partition=panda
#SBATCH --nodes=1
#SBATCH --job-name=star
#SBATCH --mem=60G

source ~/.bash_profile^C
spack load /ndj6yar star@2.7.0e%gcc@6.3.0 arch=linux-centos7-x86_64

j=1
for i in {60..62};do
file1=/athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_1.fastq.gz;
file2=/athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_2.fastq.gz;

echo "Using file:" $file1 $file2;

STAR --runMode alignReads --runThreadN 8 --genomeDir /athena/angsd/scratch/yuc4009/refdata/star_index --readFilesIn $file1 $file2 --outFileNamePrefix NRG1_${j}. --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate;

let j+=1;
done
```

#### star_enz_nrg1.bash

```{bash eval=F}
#! /bin/bash -l

#SBATCH --partition=panda
#SBATCH --nodes=1
#SBATCH --job-name=star
#SBATCH --mem=60G

source ~/.bash_profile^C
spack load /ndj6yar star@2.7.0e%gcc@6.3.0 arch=linux-centos7-x86_64

j=1
for i in {66..68};do
file1=/athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_1.fastq.gz;
file2=/athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_2.fastq.gz;

echo "Using file:" $file1 $file2;

STAR --runMode alignReads --runThreadN 8 --genomeDir /athena/angsd/scratch/yuc4009/refdata/star_index --readFilesIn $file1 $file2 --outFileNamePrefix Enz+NRG1_${j}. --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate;

let j+=1;
done
```

#### star_enz.bash

```{bash eval=F}
#! /bin/bash -l

#SBATCH --partition=panda
#SBATCH --nodes=1
#SBATCH --job-name=star
#SBATCH --mem=60G

source ~/.bash_profile^C
spack load /ndj6yar star@2.7.0e%gcc@6.3.0 arch=linux-centos7-x86_64

j=1
for i in {63..65};do
file1=/athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_1.fastq.gz;
file2=/athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_2.fastq.gz;

STAR --runMode alignReads --runThreadN 8 --genomeDir /athena/angsd/scratch/yuc4009/refdata/star_index --readFilesIn $file1 $file2 --outFileNamePrefix Enz_${j}. --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate;

let j+=1;
done
```

#### star_dmso.bash

```{bash eval=F}
#! /bin/bash -l

#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --job-name=star
#SBATCH --mem=60G

source ~/.bash_profile^C
spack load /ndj6yar star@2.7.0e%gcc@6.3.0 arch=linux-centos7-x86_64

j=1
for i in {57..59};do
file1=/athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_1.fastq.gz;
file2=/athena/angsd/scratch/yuc4009/NRG1/SRR114706${i}_2.fastq.gz;

STAR --runMode alignReads --runThreadN 8 --genomeDir /athena/angsd/scratch/yuc4009/refdata/star_index --readFilesIn $file1 $file2 --outFileNamePrefix DMSO_${j}. --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate;

let j+=1;
done
```

## Bam Quality Control

<span style="font-size:17px;">
By alignment, we got 12 bam files (4 samples, each samples has 3 replicates). Then we need to check the quality of the bam files to see the details of the alignment. Here, we used 2 quality control method: BamQC and RSeQC</span>

### RSeQC

<span style="font-size:17px;">
We used `RSeQC` to check read distribution. Firstly, we downloaded BED file from RSeQC website (https://sourceforge.net/projects/rseqc/files/BED/Human_Homo_sapiens/), then we run the following code to generate distribution plot.</span>

```{bash eval=F}
#! /bin/bash

#SBATCH --partition=panda
#SBATCH --job-name=rseqc
#SBATCH --nodes=1
#SBATCH --mem=16G

echo "Start RSeQC"
for i in /athena/angsd/scratch/yuc4009/bams_selected/*.bam; do
echo '$i';
read_distribution.py -i $i -r /athena/angsd/scratch/yuc4009/refdata/hg38_RefSeq.bed > ${i}.distribution.txt;
done

spack load -r py-multiqc@1.7

multiqc ./ -o ./
```

<span style="font-size:17px;">
The results are synthesised by `multiqc`,</span>

```{r rseqc, echo=FALSE, out.width = '100%'}
knitr::include_graphics("rseqc.png")
knitr::include_graphics("rseqc_1.png")
```

<span style="font-size:17px;">
From these two RSeQC plots, we can find that there are a large number of reads successfully aligned, most of the reads are CDS_Exons (no less than 50M and about 65% of the total reads). It is enough for the downstream analysis with such a huge amount of exon reads.</span>

<span style="font-size:17px;">
Meanwhile, we run `geneBodycoverage.py`, we take DMSO_1 to DMSO_3 as an example.</span>

```{r rseqc2, echo=FALSE, out.width = '100%'}
knitr::include_graphics("genebodycover.png")
```

<span style="font-size:17px;">
The coverage distribution is great and follows our expectation.</span>

### BamQC

<span style="font-size:17px;">
To practice more bam quality control methods, we also used `BamQC` this time. The GTF file we used here is the same as the one in STAR indexing section.</span>

```{bash eval=F}
#! /bin/bash

#SBATCH --partition=panda
#SBATCH --nodes=1
#SBATCH --job-name=bamqc
#SBATCH --mem=16G

export PATH=$PATH:/softlib/apps/EL7/BamQC/bin

for i in *bam; do
bamqc -o /athena/angsd/scratch/yuc4009/bamqc -f /athena/angsd/scratch/yuc4009/refdata/hg38.ncbiRefSeq.gtf -g /athena/angsd/scratch/yuc4009/refdata -t 8 $i;
done
```

<span style="font-size:17px;">After running the code above, we got 12 bamqc reports. These reports are downloaded from the cluster and then stored in my github repo ([bamqc folder](https://github.com/yuqcheng/ANGSD_project/tree/main/bamqc)). Because the multiqc does not support bamqc report, we do not present all reports here to prevent making the rmd file too complicated.</span>

## FeatureCounts

<span style="font-size:17px;">When we got high-quality bam files, what we need to do is count these aligned reads with genes. `featureCounts` is a good tool to do this job.</span>

### Run featureCounts

<span style="font-size:17px;">With these high quality bam files, finally, we used `featureCounts` to do read counting. </span>

```{bash eval=F}
spack load subread
featureCounts -a hg38.ncbiRefSeq.gtf -T 8 -f -M -o featureCounts_result.txt ./bams_selected/*.bam
```

### Summary plot

<span style="font-size:17px;">To check the featureCounts result, we downloaded the `featureCounts_result.txt.summary` via `WinSCP` and then plot it.</span>

```{r feat}
suppressPackageStartupMessages({
  library(ggplot2)
  library(tidyr)
})

read_gene <- read.csv("featureCounts_result.txt.summary", header = T, sep = '\t')
colnames(read_gene) = c("Status","DMSO_1","DMSO_2","DMSO_3","Enz_1","Enz_2","Enz_3","Enz+NRG1_1","Enz+NRG1_2","Enz+NRG1_3","NRG1_1","NRG1_2","NRG1_3")

read_gene <- gather(data = read_gene,
                    key = "samples",
                    value = "#reads", 
                    "DMSO_1","DMSO_2","DMSO_3","Enz_1","Enz_2","Enz_3","Enz+NRG1_1","Enz+NRG1_2","Enz+NRG1_3","NRG1_1","NRG1_2","NRG1_3")

read_gene <- read_gene[which(read_gene$`#reads`!=0),]
read_gene$modes <- rep('gene_count',nrow(read_gene))

g <- ggplot(read_gene, aes(fill=Status,y=`#reads`,x=samples)) + geom_bar(position="dodge", stat="identity")
g <- g + coord_flip() + theme(legend.position = "bottom") + labs(title = "featureCounts Summary")

g <- g + facet_grid(rows = vars(modes))

print(g)
```

<span style="font-size:17px;">The percentage of the assigned read looks good, the featureCounts result is brilliant and can be used for the downstream analysis.</span>

## Exploratory analysis

<span style="font-size:17px;">From this section, we could start the downstream analysis. The first thing is exploratory analysis. We downloaded `featureCounts_result.txt` and make DESeq matrix.</span>

```{r exp}
suppressPackageStartupMessages({
  library(DESeq2)
  library(magrittr)
  library(ggplot2)
})

readcounts <- read.table("featureCounts_result.txt", header = T)

colnames(readcounts) <- c(colnames(readcounts)[1:6], paste0("Veh_", c(1:3)), paste0("Enz_treated_", c(1:3)), paste0("Enz.NRG1_treated_",c(1:3)), paste0("NRG1_treated_",c(1:3)))
rownames(readcounts) <- make.names(readcounts$Geneid)

readcounts <- readcounts[,-c(1:6)]
sample_info <- DataFrame(condition = c(rep("Veh",3),rep("Enz_treated",3),rep("Enz.NRG1_treated",3),rep("NRG1_treated",3)), row.names = colnames(readcounts))

DESeq.ds <- DESeqDataSetFromMatrix(countData = readcounts,
                                   colData = sample_info,
                                   design = ~ condition)

DESeq.rlog <- rlog(DESeq.ds, blind = T)

#saveRDS(DESeq.ds, "featurecount_ds.rds")
```

<span style="font-size:17px;">Then we save the `DESeq.rlog` and `DESeq.ds` and generate dendrogram and PCA plots.</span>

```{r cor}
corr_rlog <- cor(assay(DESeq.rlog), method = "pearson")
corr_ds <- cor(assay(DESeq.ds), method = "pearson")

par(mfrow=c(1,2))
as.dist(1-corr_rlog) %>% hclust %>% plot(., labels = colnames(assay(DESeq.rlog)), main = "DESeq.rlog")
as.dist(1-corr_ds) %>% hclust %>% plot(., labels =colnames(assay(DESeq.ds)),main = "DESeq.ds")
```

```{r pca}
plotPCA(DESeq.rlog)
```

```{r pheatmap}
library("RColorBrewer")
sample_dist <- as.matrix(dist(t(assay(DESeq.rlog))))
rownames(sample_dist) <- paste(DESeq.rlog$condition)
colors <- colorRampPalette(rev(brewer.pal(9,'Blues')))(255)
pheatmap::pheatmap(sample_dist, color = colors)
```

## Figure 1C and 1D

```{r fig1cd}
suppressPackageStartupMessages({
  library(DESeq2)
  library(magrittr)
  library(ggplot2)
  library(pheatmap)
  library(RColorBrewer)
  library(dplyr)
  library(ggpubr)
  library(ggthemes)
  library(clusterProfiler)
  library(org.Hs.eg.db)
})

DESeq.ds <- readRDS('~/R Scripts/angsd-project/featurecount_ds.rds')
#DESeq.ds <- featurecount_ds

dds <- DESeq(DESeq.ds)
dds <- dds[rowSums(counts(dds)) > 1, ]

reactivate_gene <- results(dds, contrast = c("condition","DMSO", "Enz.NRG1"))
reactivate_gene <- subset(reactivate_gene, padj < 0.05 & abs(log2FoldChange) > 0)

rg <- as.data.frame(reactivate_gene)

rg$Symbol <- rownames(rg)

rg$logP <- -log10(rg$padj)

rg$Group <- 'not-significant'

rg$Group[which(rg$log2FoldChange > 2)] <- 'Down-regulated'
rg$Group[which(rg$log2FoldChange < -2)] <- 'Up-regulated'

rg$label <- ""
rg<- rg[order(rg$padj),]
up.gene <- head(rg$Symbol[which(rg$Group == 'Down-regulated')],15)
down.gene <- head(rg$Symbol[which(rg$Group == 'Up-regulated')],15)

res.top30.gene <- c(as.character(up.gene), as.character(down.gene))
rg$label[match(res.top30.gene, rg$Symbol)] <- res.top30.gene

rg$logP[which(rg$logP==Inf)] = 400

ggscatter(rg, 
          x='log2FoldChange', y='logP',
          size = 1,
          label = rg$label,
          font.label = 8,
          color = "Group",
          palette = c('#0000FF','#BBBBBB','#FF0000'),
          xlab = 'log2FC',
          ylab = '-log10P',
          repel = T) + 
  theme_base()+xlim(-10,10)+ylim(0,400) 

down <- length(which(reactivate_gene$log2FoldChange >0))
up <- length(which(reactivate_gene$log2FoldChange <0))

#reactivation sig
re_heatmap <- assay(dds[rownames(rg),])

annotation_col <- data.frame(Treatment = c("Veh","Veh","Veh","Enz","Enz","Enz","Enz*NRG1","Enz*NRG1","Enz*NRG1"))
rownames(annotation_col) <- colnames(re_heatmap[,1:9])
pheatmap(re_heatmap[,1:9], scale = 'row', color = rev(colorRampPalette(brewer.pal(10, "RdBu"))(20)), 
         legend_labels='Z-score', cluster_cols=F, 
         annotation_col = annotation_col,main='Reactivation profile',
         angle_col = 45,show_rownames = F) 


ego <- enrichGO(gene = rownames(rg), OrgDb = org.Hs.eg.db, ont = "CC", pAdjustMethod = "BH", keyType = 'SYMBOL',
                pvalueCutoff = 0.01, qvalueCutoff = 0.05)

#dotplot(ego)
barplot(ego,showCategory=20)
```

## Figure 2

<span style="font-size:17px;">Please check the code on my [Github repo](https://github.com/yuqcheng/ANGSD_project/tree/main/code).</span>

# Discussion

<span style="font-size:17px;">
During this final project, I find bam file quality control and downstream analysis are the two things bother me most. In the bam file quality control process, I used `bamqc` first but forgot that bamqc report cannot be read by multiqc. So I waste a lot of time making bamqc (but it is a good way to practice different qc software!). And for the downstream analysis, I got some different result with the original paper. (e.g. They said that the number of co-regulated genes is around 1900 but I got 5700.) I tried different parameters but the result did not change. I think there are many reasons that lead me to a different result. The first one is the gtf files we used are different, which would make our bam files and featureCounts result be different. The other reason might be the different analysis parameters we used. Because the author did not give his code, I cannot find the exact reason, but my overall results are corresponding with his.</span>

<span style="font-size:17px;">Luckily, my fastq files are of good quailty, so I can easily process them without trimming. But I did consider each step carefully. The whole project helps me review the whole process of the bulk RNA-seq analysis. A wondeful final project!</span>

<span style="font-size:17px;">In addition, the results of this project still have room for further analysis. In Figure 2 we found that AR and NRG1 downstream pathways have intersections, so we can further analyze these intersections. There may be potential drug targets in these co-regulated genes that cause the AR and NRG1 pathways to be closed or greatly weakened at the same time.</span>

<span style="font-size:17px;">All in all, our research proves that the hypothesis we put forward is correct. Thanks to Friederike, Luce and Merv for their work this semester, a great semester!</span>

# Key data sets

| Dataset | Path |
|:---:|:---:|
| 12 bam files | /athena/angsd/scratch/yuc4009/bams_selected |
| 2 featureCounts result | /athena/angsd/scratch/yuc4009/featureCounts |
| All other files I uploaded to my Github | https://github.com/yuqcheng/ANGSD_project |

# Reference

[1]Watson PA, Arora VK, Sawyers CL. Emerging mechanisms of resistance to androgen receptor inhibitors in prostate cancer. Nat Rev Cancer. 2015;15(12):701-711. doi:10.1038/nrc4016

[2]Antonarakis ES, Lu C, Wang H, et al. AR-V7 and resistance to enzalutamide and abiraterone in prostate cancer. N Engl J Med. 2014;371(11):1028-1038. doi:10.1056/NEJMoa1315815

[3]Ku SY, Rosario S, Wang Y, et al. Rb1 and Trp53 cooperate to suppress prostate cancer lineage plasticity, metastasis, and antiandrogen resistance. Science. 2017;355(6320):78-83. doi:10.1126/science.aah4199

[4]Bluemn EG, Coleman IM, Lucas JM, et al. Androgen Receptor Pathway-Independent Prostate Cancer Is Sustained through FGF Signaling. Cancer Cell. 2017;32(4):474-489.e6. doi:10.1016/j.ccell.2017.09.003

[5]Klemm F, Joyce JA. Microenvironmental regulation of therapeutic response in cancer. Trends Cell Biol. 2015;25(4):198-213. doi:10.1016/j.tcb.2014.11.006

[6]Zhang Z, Karthaus WR, Lee YS, et al. Tumor Microenvironment-Derived NRG1 Promotes Antiandrogen Resistance in Prostate Cancer. Cancer Cell. 2020;38(2):279-296.e9. doi:10.1016/j.ccell.2020.06.005
